crates/ruff_notebook/src/notebook.rs

```diff
@@ -388,6 +388,21 @@ impl Notebook {
         &self.cell_offsets
     }
 
+    /// Returns the start offset of the cell at index `cell` in the concatenated
+    /// text document.
+    pub fn cell_offset(&self, cell: OneIndexed) -> Option<TextSize> {
+        self.cell_offsets.get(cell.to_zero_indexed()).copied()
+    }
+
+    /// Returns the text range in the concatenated document of the cell
+    /// with index `cell`.
+    pub fn cell_range(&self, cell: OneIndexed) -> Option<TextRange> {
+        let start = self.cell_offsets.get(cell.to_zero_indexed()).copied()?;
+        let end = self.cell_offsets.get(cell.to_zero_indexed() + 1).copied()?;
```

Comment:
Checking that this actually handles the last cell properly? We store a trailing offset?
(I assume this will be elucidated elsewhere...)

Reply:
`CellOffset` contains:

* the offset `0` (even if the notebook has no cells)
* the end offset of every cell

Let's assume the notebook has one cell where the cell's length is 20 bytes. The build cell offsets then contains `[0, 20]`. Retrieving the range for cell 0 returns `0..20`. 

The function returns `None` if called with an out-of-bounds cell. E.g. `cell_range(1)` succeeds to look up the start offset (20), but it fails to retrieve the end offset and returns `None` instead